File: articles/trade-inheritance-surveying-and-practical-calculation/java/ProportionalShare.java
public class ProportionalShare {
    public static void main(String[] args) {
        double total = 1200.0;
        double[] weights = {2.0, 1.0, 1.0};
        double sum = 0.0;
        for (double w : weights) sum += w;
        for (int i = 0; i < weights.length; i++) {
            System.out.printf("share_%d=%.4f%n", i + 1, total * weights[i] / sum);
        }
    }
}

File: articles/trade-inheritance-surveying-and-practical-calculation/cpp/proportional_share.cpp
#include <iostream>
#include <vector>
#include <numeric>

int main() {
    double total = 1200.0;
    std::vector<double> weights = {2.0, 1.0, 1.0};
    double sum = std::accumulate(weights.begin(), weights.end(), 0.0);
    for (std::size_t i = 0; i < weights.size(); ++i) {
        std::cout << "share_" << i + 1 << "=" << total * weights[i] / sum << "\n";
    }
    return 0;
}
